crates/breez-sdk/core/src/sdk.rs

```diff
@@ -1004,6 +950,123 @@ impl BreezSdk {
+        let transfer = self
+            .spark_wallet
+            .transfer(request.prepare_response.amount_sats, &spark_address)
+            .await?;
+        Ok(SendPaymentResponse {
+            payment: transfer.try_into()?,
+        })
+    }
+
+    async fn send_bolt11_invoice(
+        &self,
+        invoice_details: &Bolt11InvoiceDetails,
+        spark_transfer_fee_sats: Option<u64>,
+        lightning_fee_sats: u64,
+        request: &SendPaymentRequest,
+    ) -> Result<SendPaymentResponse, SdkError> {
+        let amount_to_send = match invoice_details.amount_msat {
+            // We are not sending amount in case the invoice contains it.
+            Some(_) => None,
+            // We are sending amount for zero amount invoice
+            None => Some(request.prepare_response.amount_sats),
+        };
+        let (prefer_spark, return_pending_after_secs) = match request.options {
+            Some(SendPaymentOptions::Bolt11Invoice {
+                prefer_spark,
+                return_pending_after_secs,
+            }) => (prefer_spark, return_pending_after_secs),
+            _ => (self.config.prefer_spark_over_lightning, None),
+        };
+        let fee_sats = match (prefer_spark, spark_transfer_fee_sats, lightning_fee_sats) {
+            (true, Some(fee), _) => fee,
+            _ => lightning_fee_sats,
+        };
+
+        let payment_response = self
+            .spark_wallet
+            .pay_lightning_invoice(
+                &invoice_details.invoice.bolt11,
+                amount_to_send,
+                Some(fee_sats),
+                prefer_spark,
+            )
+            .await?;
+        let payment = match payment_response.lightning_payment {
+            Some(lightning_payment) => {
+                let ssp_id = lightning_payment.id.clone();
+                let payment = Payment::from_lightning(
+                    lightning_payment,
+                    request.prepare_response.amount_sats,
+                    payment_response.transfer.id.to_string(),
+                )?;
+                self.poll_lightning_send_payment(&payment, ssp_id);
+                payment
+            }
+            None => payment_response.transfer.try_into()?,
+        };
+
+        // Short-circuit and return the pending payment immediately.
+        if let Some(secs) = return_pending_after_secs
+            && secs == 0
+        {
+            return Ok(SendPaymentResponse { payment });
+        }
+
+        let fut = self.wait_for_payment(WaitForPaymentRequest {
+            identifier: WaitForPaymentIdentifier::PaymentId(payment.id.clone()),
+        });
+        let payment = if let Some(secs) = return_pending_after_secs {
+            match timeout(Duration::from_secs(secs.into()), fut).await {
+                Ok(res) => res?.payment,
+                // On timeout return the pending payment.
+                Err(_) => payment,
+            }
+        } else {
+            fut.await?.payment
```

Comment:
I think we should have our recommended default here. Seems to me 10 seconds to wait by default before returning the pending is reasonable. @kingonly what do you think?

Reply:
I was thinking about that, and I think a 10 second timeout doesn't make sense. Developers may not run into the 10 second timeout while developing and will assume payments are complete when this function returns. It's a separate case to account for, but it's kind-of hidden. So I would opt for either returning a pending payment immediately, or waiting for the payment to complete, but not something in the middle.

Reply:
Seems a bit overkill?
Why would some one want to wait for infinity? When sending payment I think one always want some timeout. If tracking is needed then using events is better.